public/app/plugins/panel/graph3/plugins/AnnotationsPlugin.tsx

```diff
@@ -28,52 +27,27 @@ export const AnnotationsPlugin: React.FC<AnnotationsPluginProps> = ({ annotation
     [timeZone]
   );
 
-  const annotationsData = useMemo(() => {
-    return getAnnotationsFromData(annotations);
-  }, [annotations]);
+  const [annotationEvents, setAnnotationEvents] = useState<AnnotationEvent[]>([]);
 
-  const annotationMarkers = useMemo(() => {
-    if (!plotContext || !plotContext?.u) {
-      return null;
+  useEffect(() => {
+    let subscription: Subscription;
+    if (plotCtx.isPlotReady) {
+      subscription = getAnnotationsFromData(annotations).subscribe(result => {
+        console.log(result);
+        setAnnotationEvents(result);
+      });
     }
-    const markers = [];
-
-    for (let i = 0; i < annotationsData.length; i++) {
-      const annotation = annotationsData[i];
-      if (!annotation.time) {
-        continue;
+    return () => {
+      if (subscription) {
+        subscription.unsubscribe();
       }
```

Comment:
@hugohaggmark I've replaced useObservable with effect. I don't like the useObservable neither, it's a one-off hook :/

Reply:
Awesome!

Reply:
What do you mean one-off? it adds the observable to the dependency list

Reply:
IMO I am not sure why we need observable here, we should be able to just use the DataFrame results directly. See no reason for figuring out what fields to use to be async

Reply:
Will make using annotations in other visualizations unnecessarily hard / complex. Not sure what the future is for AnnotationEvent type, thinking is that it might go away and just use a DataFrame directly.

Reply:
not longer using observables after https://github.com/grafana/grafana/pull/28071/commits/d33afca8eb6802bb9013132bf260998e367a5624 :)